### PR TITLE
Include resource to create db instance from snapshot

### PR DESCRIPTION
**Pull request by WinnyTroy**

### CODE REVIEW

File: storage.tf [hunk 1]
```diff
@@ -31,6 +33,35 @@ resource "aws_db_instance" "main" {
...
 }
 
+resource "aws_db_instance" "from-snapshot" {
+  count = length(var.postgresql_source_snapshot_identifier) == 0 ? 0 : 1
+
+  apply_immediately      = var.postgresql_apply_immediately
+  identifier             = var.postgresql_name
+  storage_type           = var.postgresql_storage_type
+  instance_class         = var.postgresql_instance_class
+  username               = var.postgresql_username
```
Comment (jasonrogena) [inline storage.tf]: @WinnyTroy @DavisRayM could we confirm that we need the username and password variables for RDS instances that are being created from snapshots.

Comment (DavisRayM) [reply storage.tf]: The [username](https://www.terraform.io/docs/providers/aws/r/db_instance.html#username) and [password](https://www.terraform.io/docs/providers/aws/r/db_instance.html#password) are not required for databases built from snapshots. They utilize the originals username and password.

CC: @WinnyTroy

Comment (WinnyTroy) [reply storage.tf]: Cool, I'll remove those values from this resource

**Review by DavisRayM — APPROVED** [review]
LGTM, lets have @jasonrogena review though.

**Review by jasonrogena — APPROVED** [review]

**Pull request merged**